packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -445,6 +445,21 @@ class EditableTextState extends State<EditableText> implements TextInputClient {
     _scrollController.jumpTo(_getScrollOffsetForCaret(caretRect));
   }
 
+  bool _textChangedSinceLastCaretUpdate = false;
+
+  void _handleCaretChanged(Rect caretRect) {
+    // If the caret location has changed due to an update to the text or
+    // selection, then scroll the caret into view.
+    if (_textChangedSinceLastCaretUpdate) {
+      _textChangedSinceLastCaretUpdate = false;
+      _scrollController.animateTo(
+        _getScrollOffsetForCaret(caretRect),
+        curve: Curves.linear,
```

Comment:
Let's try `Curves.fastOutSlowIn` and see how it feels. That's the usual material curve.